openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -34,22 +34,46 @@ class InvalidNotFound(NotFound):
     default_detail = "Invalid XBlock key"
 
 
+def _block_type_overrides(request_args):
+    """
+    Returns the block_type_overrides dict created from the request parameters:
+
+    * from_block_types: list of block types to override
+    * to_block_types: list of block types to use as overrides
+    """
+    from_block_types = request_args.get("from_block_types", "")
+    to_block_types = request_args.get("to_block_types", "")
```

Comment:
@pomegranited I'm not sure about this api - it's not clear how the `from` and `to` values relate.  Both from seeing an example querystring and the docstring here.  Also then we need to define in the docs the behaviour when the two lists are not the same length.

```
?from_block_types=old1,old2,old3&to_block_types=new1,new2,new3
```

Idk it seems very brittle.

Maybe we could use something like this instead?:

```
?block_override=old1:new1&block_override=old2:new2
```

The code would need to look something like:

```python3
requested_overrides = request_args.getlist('block_overrides', [])

overrides = {}
for item in requested_overrides:
    # split on `:` (or whatever deliminator we want)
    # verify there are exactly two non-empty entries in the split list
    # overrides[old] = new

return overrides
```

Reply:
Yeah, there's pros and cons to both, and no clear convention on what to do in this case. The [edx REST API conventions](https://openedx.atlassian.net/wiki/spaces/AC/pages/18350757/edX+REST+API+Conventions#4.-URL-Parameters) want comma-separated strings for list parameters, but they don't say anything about maps.

I liked this way because it was easy to generate from a dict (since [python3 orders `keys`/`values` on insertion](https://docs.python.org/3/library/stdtypes.html#dict)), and easy to reassemble using `zip`.

We'd have the same issue with having to check parameter format and empty-strings if we use a separator like `:` too, so I'm inclined to keep this format as it was what was requested in the ticket. Will change it if the upstream reviewers object.

Reply:
From @bradenmacdonald 's [comment](https://github.com/edx/edx-platform/pull/29517#issuecomment-997115143):

> RE the query string API, my personal preference is strongly for ?block_override=old1:new1&block_override=old2:new2 but don't take that as the final word on the subject.

I'm ok with this approach.. who's the best person at Open edX to ask now that Nimisha's gone? CC @ormsbee

Reply:
NVM -- we're [classifying this as a temporary change](https://github.com/edx/edx-platform/pull/29517#issuecomment-1006174331), so aren't going to spend more time debating how the query parameters are passed in.

Reply:
NVM again -- [for security reasons](https://github.com/edx/edx-platform/pull/29517#issuecomment-1008449999), we've [hardcoded the block override types](https://github.com/open-craft/edx-platform/blob/jill/lx-2365/openedx/core/djangoapps/xblock/rest_api/views.py#L32-L36), and are using a simple boolean argument instead: `?lx_block_types=1`.